connectors/rocketmq-connect-jdbc/src/main/java/org/apache/rocketmq/connect/jdbc/dialect/impl/OracleDatabaseDialect.java

```diff
@@ -0,0 +1,308 @@
+  }
+
+  @Override
+  protected String currentTimestampDatabaseQuery() {
+    return "select CURRENT_TIMESTAMP from dual";
+  }
+
+  @Override
+  protected String checkConnectionQuery() {
+    return "SELECT 1 FROM DUAL";
+  }
+
+  @Override
+  public StatementBinder statementBinder(
+      PreparedStatement statement,
+      JdbcSinkConfig.PrimaryKeyMode pkMode,
+      SchemaPair schemaPair,
+      FieldsMetadata fieldsMetadata,
+      TableDefinition tableDefinition,
+      JdbcSinkConfig.InsertMode insertMode
+  ) {
+    return new PreparedStatementBinder(
+        this,
+        statement,
+        pkMode,
+        schemaPair,
+        fieldsMetadata,
+        tableDefinition,
+        insertMode
+    );
+  }
+
+  @Override
+  public void bindField(
+      PreparedStatement statement,
+      int index,
+      Schema schema,
+      Object value,
+      ColumnDefinition colDef
+  ) throws SQLException {
+    if (value == null) {
+      statement.setObject(index, null);
+    } else {
+      boolean bound = maybeBindLogical(statement, index, schema, value, colDef);
+      if (!bound) {
+        bound = maybeBindPrimitive(statement, index, schema, value, colDef);
+      }
+      if (!bound) {
+        throw new ConnectException("Unsupported source data type: " + schema.getFieldType());
+      }
+    }
+  }
+
+  protected boolean maybeBindPrimitive(
+      PreparedStatement statement,
+      int index,
+      Schema schema,
+      Object value,
+      ColumnDefinition colDef
+  ) throws SQLException {
+    if (colDef == null) {
+      return super.maybeBindPrimitive(statement, index, schema, value);
+    }
+
+    if (schema.getFieldType() == FieldType.STRING) {
+      if (colDef.type() == Types.CLOB) {
+        statement.setCharacterStream(index, new StringReader((String) value));
+        return true;
+      } else if (colDef.type() == Types.NCLOB) {
+        statement.setNCharacterStream(index, new StringReader((String) value));
+        return true;
+      } else if (colDef.type() == Types.NVARCHAR || colDef.type() == Types.NCHAR) {
+        statement.setNString(index, (String) value);
+        return true;
+      } else {
+        return super.maybeBindPrimitive(statement, index, schema, value);
+      }
+    }
+
+    if (schema.getFieldType() == FieldType.BYTES && colDef.type() == Types.BLOB) {
+      if (value instanceof ByteBuffer) {
+        statement.setBlob(index, new ByteArrayInputStream(((ByteBuffer) value).array()));
+      } else if (value instanceof byte[]) {
+        statement.setBlob(index, new ByteArrayInputStream((byte[]) value));
+      } else {
+        return super.maybeBindPrimitive(statement, index, schema, value);
+      }
+      return true;
+    }
+    return super.maybeBindPrimitive(statement, index, schema, value);
+  }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  @Override
+  protected String getSqlType(SinkRecordField field) {
+    if (field.schemaName() != null) {
```

Comment:
useless code

Reply:
> useless code
Oracle dialect has been deleted. Submit after the test is completed